src/View/Form/ContextInterface.php

```diff
@@ -48,9 +48,14 @@ public function isCreate();
      *
      * @param string $field A dot separated path to the field a value
      *   is needed for.
+     * @param array $options Options:
+     *   - `default`: Default value to return if no value found in request
+     *     data or context record.
+     *   - `schemaDefault`: Boolen indicating whether default value from
+     *      context's schema should be used if it's not explicitly provided.
      * @return mixed
      */
-    public function val($field);
+    public function val($field, $options = []);
```

Comment:
We'll need to update the elasticsearch plugin too. This is an API breaking change unfortunately :cry: as changing the interface will cause fatal errors in userland code.